bundles/persistence/org.openhab.persistence.jdbc/java/org/openhab/persistence/jdbc/utils/StringUtilsExt.java

```diff
@@ -0,0 +1,265 @@
+		if(dbURI.getPort() != -1)props.put("portNumber", dbURI.getPort() + "");
+		if(dbURI.getHost() != null)props.put("serverName", dbURI.getHost());
+		
+		return props;
+	}
+	
+    /**
+     * @param s
+     * @param substr
+     * @return - Returns a String before the last occurrence of a Substring
+     */
+    public static String stringBeforeLastSubstr(String s, String substr) {
+    	ArrayList<Integer> a = substrPos(s, substr);
+    	return s.substring(0, a.get(a.size()-1));
+    } 
+    
+    /**
+     * @param s
+     * @param substr
+     * @return - Returns a String after the last occurrence of a Substring
+     */
+    public static String stringAfterLastSubstr(String s, String substr) {
+    	ArrayList<Integer> a = substrPos(s, substr);
+    	return s.substring(a.get(a.size()-1) + 1);
+    } 
+
+    /**
+     * @param s
+     * @param substr
+     * @return - Returns a String after the first occurrence of a Substring
+     */
+    public static String stringAfterSubstr(String s, String substr) {
+    	return s.substring(s.indexOf(substr) + 1);
+    }   
+    
+    /**
+     * @param s
+     * @param substr
+     * @param pos
+     * @return - Returns a String after the n occurrence of a Substring
+     */
+    public static String stringAfterSubstr(String s, String substr, int n) {
+    	return s.substring(substrPos(s, substr).get(n) + 1);
+    }
+    
+    /**
+     * @param s
+     * @param substr
+     * @return - Returns a String before the first occurrence of a Substring
+     */
+    public static String stringBeforeSubstr(String s, String substr) {
+    	return s.substring(0, s.indexOf(substr));
+    }
+    
+    /**
+     * @param s
+     * @param substr
+     * @param pos
+     * @return - Returns a String before the n occurrence of a Substring
+     */
+    public static String stringBeforeSubstr(String s, String substr, int n) {
+    	return s.substring(0, substrPos(s, substr).get(n));
+    }
+
+    /**
+     * @param s
+     * @param substr
+     * @return - Returns an ArrayList with Indices of the occurrence of a Substring
+     */
+    public static ArrayList<Integer> substrPos(String s, String substr) {
+    	return substrPos(s, substr, true);
+    }
+    /**
+     * @param s
+     * @param substr
+     * @param ignoreCase
+     * @return - Returns an ArrayList with Indices of the occurrence of a Substring
+     */
+    public static ArrayList<Integer> substrPos(String s, String substr, boolean ignoreCase) {
+        int substrLength = substr.length();
+        int strLength = s.length();
+        ArrayList<Integer> arr = new ArrayList<Integer>();
+
+        for(int i = 0; i < strLength - substrLength + 1; i++) {
+            if(s.regionMatches(ignoreCase, i, substr, 0, substrLength))  {
+            	arr.add(i);
+            }
+        }
+        return arr;
+    }
+    
+    public static void main(String[] args) throws Exception {
```

Comment:
Is this main() method for testing purposes? In that case, could it be moved to an unit test? I believe we have junit available.